### ISSUE TITLE
Borderless form request

### ISSUE THREAD

**Issue by ericsia** [Onelio/Win32GUI#6]
Can you add borderless form option?

**Comment by moyamejiasr** [comment]
Try
 ```
allowTitleBar(false);
```

**Comment by ericsia** [comment]
> allowTitleBar(false);

yes it can but can't move the entire form anymore.
Can you update it so that the borderless form can draggable and move?
ReleaseCapture()
SendMessage(Handle, ..., ..., 0)
https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-sendmessage

**Comment by moyamejiasr** [comment]
Then also activate
`setDraggable(true);`

**Comment by ericsia** [comment]
> Then also activate
> `setDraggable(true);`

wow I didn't know it was there. It would be nice if you provide documentation of the features you implemented.

How about change the form color? is it this one?
window.setBackground

what is the parameter? Say if I want color other than white grey, black, is it possible? say red

**Comment by moyamejiasr** [comment]
Hi, I've not provided any documentation already because I'm planning a lot of refactoring for whenever I have time to. So many things may change in the future.
About your question, you can use setBackground and provide it with any function from here: https://github.com/Onelio/Win32GUI/blob/master/Win32/Manager.cpp#L19 for example: setBackground(Brush(Hex("#FFFFFF")));

Keep in mind that (with how it works at the moment) some controls may need to be applied the same background or at least setBackground(Invisible);

**Comment by ericsia** [comment]
> setBackground(Brush(Hex("#FFFFFF")));

owh I didn't know you provided Brush(Hex("#FFFFFF"));
That would be super good enough, you see, that is the problem, people unrealised there is smt ald there.

Also can you update the code so that it support langeuage other than english?
Like I set the label to chinese it became unprintable.
chinese character is 2 char, english ascii is 1 char

So this is the size, here is a test:
TextView lbl(&window, "下次运行将文件输出到源文件目录");

also it will be more helpful if you can inherit the background color if it is changed
Also how to set the text color? lbl.setTextColor(???);
![imagine2](https://user-images.githubusercontent.com/24624808/104837057-28353080-58ed-11eb-9c83-3a688b44aca5.png)

**Comment by moyamejiasr** [comment]
Yes, that is because it is set to CHAR instead if WIDE CHAR, which is one of the many things that I'm planning to change in the future to allow the two methods depending on the VSTUDIO build. For the moment you can either not write with wide characters or update every char, string and the like to wchar, wstring, etc...
For the colour inheritance issue: I'm also planning that but sadly it is not available at the moment. It is a big project and I need to make a lot of research for many things to work. You can either change every control background-colour after the creation as I said with invisible or the colour that you want to use.

I'll start working on it very soon, I plan on recreating a lot of stuff to make it more reliable and friendly with other input methods. But that will have to wait a few days more. I also plan to add a "dark mode" based in the system settings and more but it will take some time to finish.

**Comment by ericsia** [comment]
> Yes, that is because it is set to CHAR instead if WIDE CHAR, which is one of the many things that I'm planning to change in the future to allow the two methods depending on the VSTUDIO build. For the moment you can either not write with wide characters or update every char, string and the like to wchar, wstring, etc...
> For the colour inheritance issue: I'm also planning that but sadly it is not available at the moment. It is a big project and I need to make a lot of research for many things to work. You can either change every control background-colour after the creation as I said with invisible or the colour that you want to use.
> 
> I'll start working on it very soon, I plan on recreating a lot of stuff to make it more reliable and friendly with other input methods. But that will have to wait a few days more.

I will be excited of it, Write down the feature as you implement it.
I love your work, this is a true OOP approached.
So people can refer.
How to set the text color? lbl.setTextColor(?);

**Comment by moyamejiasr** [comment]
Yes, for text color you use setTextColor(Hex("")); but keep in mind that setTextColor and setBackground may not work with every control. It is not an issue with the OOP approach but with the WINAPI, because some controls do not allow those changes. I also plan to "workaround" that but that also needs more research 😔

**EDIT**: setTextColor does not use the Brush function, that is because setBackground also supports pictures so you need to usually convert it but since setTextColor does not you don't actually need it.

**Comment by ericsia** [comment]
> Yes, for text color you use setTextColor(Hex("")); but keep in mind that setTextColor and setBackground may not work with every control. It is not an issue with the OOP approach but with the WINAPI, because some controls do not allow those changes. I also plan to "workaround" that but that also needs more research 😔
> 
> EDIT: setTextColor does not use the Brush function, that is because setBackground also supports pictures so you need to usually convert it but since setTextColor does not you don't actually need it.

Hmm, I think there's a bug
setTextColor(Hex(""));
cause the text size shrink
Edit: my bad, I set the size too small
![imagine2](https://user-images.githubusercontent.com/24624808/104837527-1903b200-58f0-11eb-9edd-594d60ada244.png)

**Comment by moyamejiasr** [comment]
I'll check into that later if the issue persists, but you should actually specify a color when using it, like "#FFFFFF". Also if you change the text color after showing the form try to control.redraw(); to update the content. Seems like I forgot to include that in my code :P

If the problem continues, send me by email your code (or at least the parts related to the UI) for me to review. You can find it in my profile.

**Comment by ericsia** [comment]
> control.redraw();

my bad, I thought it is same as 10pt so I set lbl.setTextSize(10); no wonder it became so small.
I've changed it to 20 now it's large enough.

Btw since it's an oop approach, just a suggestion, take in the hex value only so it don't cause confusion.
You know what I mean?
lbl.setHexBackgroundColor("#FFFFFF");
window.setHexBackgroundColor("#FFFFFF");

Also one suggestion, maybe have something like setCursorHand() coz this will be helpful
![imagine2](https://user-images.githubusercontent.com/24624808/104839733-fd9fa380-58fd-11eb-8f2a-7eca40577e5d.png)
or you implemented it already?
1. how to set the cursor become hand?
2. How to use setOnHover? I am planning to change the color on Hover
3. how to change button background color?  button1.setBackground(Brush(Hex(...))); seem not working
4. How to paint another color to the form?
![imagine2](https://user-images.githubusercontent.com/24624808/104842915-de574500-5902-11eb-9639-a1a9a0639efc.png)

**Comment by moyamejiasr** [comment]
Yes, If you want to implement a custom button I recommend using a PictureBox with on hover because buttons cannot be changed in C++ (even tho you can safely do it in C#). Also, I don't have the cursor thing implemented rn but if you change my code, the execute at PictureBox.cpp to detect WM_SETCURSOR and send SetCursor it should work just fine.
Or you could play with ImageButton and see if it works for you.

About the setOnHover you need to create a function with void myfunc(Control*, bool) and if true, the cursor is on top of the item.

Finally, if you want to change the form color just call the set background function again.

**Comment by ericsia** [comment]
> Yes, If you want to implement a custom button I recommend using a PictureBox with on hover because buttons cannot be changed in C++ (even tho you can safely do it in C#). Also, I don't have the cursor thing implemented rn but if you change my code, the execute at PictureBox.cpp to detect WM_SETCURSOR and send SetCursor it should work just fine.
> Or you could play with ImageButton and see if it works for you.
> 
> About the setOnHover you need to create a function with void myfunc(Control*, bool) and if true, the cursor is on top of the item.
> 
> Finally, if you want to change the form color just call the set background function again.

no I mean how can we paint a certain portion on window form?
like in C# we can set the coordinate and the rectangle size to paint the form. Like paint the orange region only.

Also I mailed you a sample button c++ code with color, see if you got smt

**Comment by moyamejiasr** [comment]
I have no idea about that, I guess you could implement your own DRAW call, maybe I'll add that feature in the future but I never made the project with that option in mind. 
And thanks for the sample, I'll check it later.
